<launch>

    <arg name="robot_name" default="sam"/>

    <node name="dyn_alt_pid_1" pkg="pid" type="controller" ns="dynamic_alt" output="screen" >
      <param name="Kp" value="2.0" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="0.0" />
      <param name="upper_limit" value="0.5" />
      <param name="lower_limit" value="-0.5" />
      <param name="windup_limit" value="100" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
      <!--CHANGE LAUNCH FILE TO CHOOSE SUBSCRIBER AND PUBLISHER TOPICS!!/-->
      <!--UPDATE ALTITUDE TOPIC!!/-->
      <param name="topic_from_plant" value="/$(arg robot_name)/dr/alt" />
      <param name="topic_from_controller" value="pitch_setpoint" />
      <remap from="setpoint" to="setpoint" />
      <!--remap from="pitch_setpoint" to="/pitch_setpoint" /-->
      <!--remap from="/plant_state_pitch" to="state" />
      <remap from="control_effort" to="/control_effort" /-->
      <remap from="pid_enable" to="pid_enable" />
     </node>

     <node name="dyn_alt_pid_2" pkg="pid" type="controller" ns="dynamic_alt" output="screen" >
       <param name="Kp" value="0.5" />
       <param name="Ki" value="0.0" />
       <param name="Kd" value="0.0" />
       <param name="upper_limit" value="0.2" />
       <param name="lower_limit" value="-0.15" />
       <param name="windup_limit" value="100" />
       <param name="max_loop_frequency" value="100.0" />
       <param name="min_loop_frequency" value="100.0" />
       <!--CHANGE LAUNCH FILE TO CHOOSE SUBSCRIBER AND PUBLISHER TOPICS!! /-->
       <param name="topic_from_plant" value="/$(arg robot_name)/dr/pitch" />
       <param name="topic_from_controller" value="/$(arg robot_name)/ctrl/elevator_setpoint" />
       <remap from="setpoint" to="pitch_setpoint" />
       <!--remap from="/plant_state_pitch" to="state" />
       <remap from="control_effort" to="/control_effort" /-->
       <remap from="pid_enable" to="pid_enable" />
      </node>



</launch>
